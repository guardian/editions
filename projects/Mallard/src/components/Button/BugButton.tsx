import React from 'react'
import { Button } from './Button'
import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
    button: {
        position: 'absolute',
        bottom: 20,
        right: 20,
        zIndex: 9999,
    },
})

const BugButton = ({ onPress }: { onPress: () => void }) => (
    <Button
        accessibilityLabel="Report a bug button"
        accessibilityHint="Opens a dialog asking if you want to include diagnostic information to your report"
        style={styles.button}
        onPress={onPress}
        alt="Report a bug"
        icon="î‚š"
    />
)

export { BugButton }
