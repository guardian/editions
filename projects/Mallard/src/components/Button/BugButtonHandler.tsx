import React, { useContext } from 'react';
import { AccessContext } from 'src/authentication/AccessContext';
import { createMailtoHandler } from 'src/helpers/diagnostics';
import { isInBeta } from 'src/helpers/release-stream';
import { DIAGNOSTICS_TITLE } from 'src/helpers/words';
import { useNetInfoProvider } from 'src/hooks/use-net-info-provider';
import { BugButton } from './BugButton';

const BugButtonHandler = () => {
	const { attempt } = useContext(AccessContext);
	const {
		isConnected,
		isPoorConnection,
		downloadBlocked,
		isInternetReachable,
		type,
	} = useNetInfoProvider();
	const netInfo = {
		isConnected,
		isPoorConnection,
		downloadBlocked,
		isInternetReachable,
		type,
	};
	return isInBeta() ? (
		<BugButton
			onPress={createMailtoHandler(
				'Report a bug',
				'',
				attempt,
				netInfo,
				DIAGNOSTICS_TITLE,
			)}
		/>
	) : null;
};

export { BugButtonHandler };
