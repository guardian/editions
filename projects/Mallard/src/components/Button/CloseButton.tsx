import React from 'react';
import type { ArticlePillar } from '../../../../Apps/common/src';
import { Button, ButtonAppearance } from './Button';

const CloseButton = ({
	onPress,
	appearance,
	accessibilityLabel,
	accessibilityHint,
	pillar,
}: {
	onPress: () => void;
	appearance?: ButtonAppearance;
	accessibilityLabel: string;
	accessibilityHint: string;
	pillar?: ArticlePillar;
}) => {
	return (
		<Button
			icon={'\uE04F'}
			alt="Dismiss"
			appearance={appearance ?? ButtonAppearance.Default}
			onPress={onPress}
			accessibilityLabel={accessibilityLabel}
			accessibilityRole="button"
			accessibilityHint={accessibilityHint}
			pillar={pillar ?? null}
		/>
	);
};

export { CloseButton };
