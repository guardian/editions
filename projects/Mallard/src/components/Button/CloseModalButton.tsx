import React from 'react'
import { StyleSheet } from 'react-native'
import { Button } from './Button'
import { color } from 'src/theme/color'

const closeModalButtonStyles = (bgColor?: string, borderColor?: string) =>
    StyleSheet.create({
        dismissText: {
            color: bgColor ? 'white' : color.primary,
        },
        dismissButton: {
            paddingHorizontal: 0,
            backgroundColor: bgColor ? bgColor : 'transparent',
            borderColor: borderColor ? borderColor : color.primary,
            borderWidth: 1,
        },
    })

const CloseModalButton = ({
    onPress,
    bgColor,
    borderColor,
}: {
    onPress: () => void
    bgColor?: string
    borderColor?: string
}) => {
    return (
        <Button
            icon=""
            alt="Dismiss"
            buttonStyles={[
                closeModalButtonStyles(bgColor, borderColor).dismissButton,
            ]}
            textStyles={
                closeModalButtonStyles(bgColor, borderColor).dismissText
            }
            onPress={onPress}
        />
    )
}

export { CloseModalButton }
