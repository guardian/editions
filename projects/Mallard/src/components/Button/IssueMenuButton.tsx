import React from 'react';
import { Button } from './Button';

const IssueMenuButton = ({ onPress }: { onPress: () => void }) => (
	<Button icon={'\uE04A'} alt="More issues" onPress={onPress} />
);

export { IssueMenuButton };
