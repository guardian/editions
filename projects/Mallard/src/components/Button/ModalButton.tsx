import React from 'react';
import type { StyleProp, TextStyle, ViewStyle } from 'react-native';
import { StyleSheet } from 'react-native';
import { Button, ButtonAppearance } from './Button';

const styles = StyleSheet.create({
	button: {
		marginTop: 10,
		marginRight: 10,
	},
});

const ModalButton = (props: {
	onPress: () => void;
	children: string;
	alt?: string;
	buttonStyles?: StyleProp<ViewStyle>;
	textStyles?: StyleProp<TextStyle>;
	buttonAppearance?: ButtonAppearance;
}) => (
	<Button
		{...props}
		alt={props.alt || props.children}
		style={styles.button}
		buttonStyles={props.buttonStyles}
		textStyles={props.textStyles}
		appearance={props.buttonAppearance || ButtonAppearance.light}
	/>
);

export { ModalButton };
