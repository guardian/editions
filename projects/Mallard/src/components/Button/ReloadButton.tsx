import React from 'react';
import { View } from 'react-native';
import { useInsets } from 'src/hooks/use-screen';
import { Button, ButtonAppearance } from './Button';

export const ReloadButton: React.FC<{
	onPress: () => void;
}> = ({ onPress }) => {
	const { top, left } = useInsets();
	return (
		<View
			style={[
				{
					position: 'absolute',
					top: top + 20,
					right: 70,
					zIndex: 99999,
				},
			]}
		>
			<Button
				appearance={ButtonAppearance.Tomato}
				onPress={onPress}
				buttonStyles={{ left: 0 }}
			>
				Reload
			</Button>
		</View>
	);
};
