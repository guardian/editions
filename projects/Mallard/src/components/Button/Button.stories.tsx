import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { BugButton } from './BugButton';
import { ButtonAppearance } from './Button';
import { CloseButton } from './CloseButton';
import { IssueMenuButton } from './IssueMenuButton';
import { ModalButton } from './ModalButton';
import { ReloadButton } from './ReloadButton';
import { SettingsButton } from './SettingsButton';

storiesOf('Buttons', module)
	.addDecorator(withKnobs)
	.add('IssueMenuButton - default', () => (
		<IssueMenuButton onPress={() => {}} />
	))
	.add('CloseButton - default', () => {
		return (
			<CloseButton
				onPress={() => {}}
				appearance={ButtonAppearance.default}
				accessibilityHint="Accesibility Hint"
				accessibilityLabel="Accesibility Label"
			/>
		);
	})
	.add('CloseButton - with appearance', () => {
		return (
			<CloseButton
				onPress={() => {}}
				appearance={ButtonAppearance.modal}
				accessibilityHint="Accesibility Hint"
				accessibilityLabel="Accesibility Label"
			/>
		);
	})
	.add('BugButton - default', () => <BugButton onPress={() => {}} />)
	.add('ReloadButton - default', () => <ReloadButton onPress={() => {}} />)
	.add('ModalButton - default', () => (
		<ModalButton onPress={() => {}}>
			{text('Button Text', 'Sign In')}
		</ModalButton>
	))
	.add('SettingsButton - default', () => (
		<SettingsButton onPress={() => {}} />
	));
