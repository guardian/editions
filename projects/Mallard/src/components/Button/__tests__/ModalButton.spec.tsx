import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { ModalButton } from '../ModalButton'
import { ButtonAppearance } from '../Button'

describe('ModalButton', () => {
    it('should display a default ModalButton', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <ModalButton onPress={() => {}}>Sign out</ModalButton>,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a ModalButton with accessible text', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <ModalButton onPress={() => {}} alt="This button signs you out">
                Sign out
            </ModalButton>,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a ModalButton with an alternative appearance', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <ModalButton
                onPress={() => {}}
                buttonAppearance={ButtonAppearance.apricot}
            >
                Sign out
            </ModalButton>,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
