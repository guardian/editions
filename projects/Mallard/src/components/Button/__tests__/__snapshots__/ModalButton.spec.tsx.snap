// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalButton should display a ModalButton with accessible text 1`] = `
<View
  accessibilityHint="This button signs you out"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GuardianTextSans-Regular",
            "fontSize": 17,
            "lineHeight": 21,
          },
          Object {
            "color": "#121212",
          },
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextSans-Bold",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Object {
              "textAlign": "auto",
            },
            Object {
              "color": "#052962",
            },
            undefined,
          ],
        ]
      }
    >
      Sign out
    </Text>
  </View>
</View>
`;

exports[`ModalButton should display a ModalButton with an alternative appearance 1`] = `
<View
  accessibilityHint="Sign out"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        Object {
          "backgroundColor": "#FF7F0F",
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GuardianTextSans-Regular",
            "fontSize": 17,
            "lineHeight": 21,
          },
          Object {
            "color": "#121212",
          },
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextSans-Bold",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Object {
              "textAlign": "auto",
            },
            Object {
              "color": "#ffffff",
            },
            undefined,
          ],
        ]
      }
    >
      Sign out
    </Text>
  </View>
</View>
`;

exports[`ModalButton should display a default ModalButton 1`] = `
<View
  accessibilityHint="Sign out"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GuardianTextSans-Regular",
            "fontSize": 17,
            "lineHeight": 21,
          },
          Object {
            "color": "#121212",
          },
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextSans-Bold",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Object {
              "textAlign": "auto",
            },
            Object {
              "color": "#052962",
            },
            undefined,
          ],
        ]
      }
    >
      Sign out
    </Text>
  </View>
</View>
`;
